photosynthesis_metrics/kid.py

```diff
@@ -212,49 +148,117 @@ def _sqn(tensor : torch.Tensor) -> torch.Tensor:
 
 
 class KID(BaseFeatureMetric):
-    r"""Creates a criterion that measures Kernel Inception Distance (polynomial MMD) for two datasets of images.
+    r"""Interface of Kernel Inception Distance.
+    It's computed for a whole set of data and uses features from encoder instead of images itself to decrease
+    computation cost. KID can compare two data distributions with different number of samples.
+    But dimensionalities should match, otherwise it won't be possible to correctly compute statistics.
 
     Args:
-        degree: Degree of a polynomial functions used in kernels. Default: 3
-        gamma: Kernel parameter. See paper for details
-        coef0: Kernel parameter. See paper for details
-        var_at_m: Kernel variance. Default is `None`
-        ret_var: Whether to return variance after the distance is computed.
-                    This function will return Tuple[float, float] in this case. Default: False
+        predicted_features: Low-dimension representation of predicted image set. Shape (N_pred, encoder_dim)
+        target_features: Low-dimension representation of target image set. Shape (N_targ, encoder_dim)
+
+    Returns:
+        score: Scalar value of the distance between image sets features.
+        variance (optional): If `ret_var` is True, also returns variance
```

Comment:
Specify types here since there are no type annotations:
```suggestion
        predicted_features(torch.Tensor): Low-dimension representation of predicted image set. Shape (N_pred, encoder_dim)
        target_features(torch.Tensor): Low-dimension representation of target image set. Shape (N_targ, encoder_dim)

    Returns:
        score(torch.Tensor): Scalar value of the distance between image sets features.
        variance(Optional[torch.Tensor]): If `ret_var` is True, also returns variance
```